config/vendor-api-v1.yml

```diff
@@ -753,6 +766,48 @@ components:
           example: University of Amsterdam
           maxLength: 256
           nullable: true
+        hesa_degtype:
+          type: string
+          description: HESA code for degree type
+          example: '001'
+          maxLength: 20
+          nullable: true
+        hesa_degsbj:
+          type: string
+          description: HESA code for degree subject
+          example: '100048'
+          maxLength: 20
+          nullable: true
+        hesa_degclss:
+          type: string
+          description: HESA code for degree class
+          example: '01'
+          maxLength: 20
```

Comment:
Shouldn't these fields all have maxlength of the legitimate data? ie in this case it would be 2

Reply:
Depends... I've treated `maxLength` as an attribute of the Apply API and different to `Length` that is part of the HESA spec. `maxLength`, from this perspective, is how large a field vendors should be allocating in their database to store this information, not necessarily the length of values received. As we don't cap the length of values returned in the presenter, the current arrangement seems the safest option. Furthermore, if HESA codes were to change in terms of length, we could easily cope with these changes in the code while keeping Apply's `maxLength` spec constant.

What do you think?

Reply:
Sure — agree it makes sense to leave the fields a bit baggy in case the coding changes.